src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx

```diff
@@ -0,0 +1,143 @@
+---
+title: Install the Python agent on Microsoft Azure Container Apps and App Service
+metaDescription: How to install APM for Python on Microsoft Azure App Services and Container Apps.
+redirects:
+  - /docs/agents/python-agent/hosting-services/python-agent-microsoft-azure
+  - /docs/python/python-agent-on-microsoft-azure
+freshnessValidatedDate: never
+---
+
+Microsoft Azure offers two platform services that manage your web or containerized apps. You can install the Python agent into both [Microsoft Azure App Service](https://learn.microsoft.com/en-us/azure/app-service/overview) and [Azure Container Apps](https://learn.microsoft.com/en-us/azure/container-apps/overview), which helps you track how Microsoft Azure manages app resource consumption, performance, and HTTP traffic. 
+
+## Compatibility and requirements [#compatibility-requirements]
+
+Before you install the Python agent, we recommend: 
+
+* Reviewing the Python agent's [compatibility and requirements doc](/docs/agents/python-agent/getting-started/compatibility-requirements-python-agent)
+* Setting up a Python app on one (or both!) of the Azure managed platforms
+* Installing the [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli-macos) in your environment
+
+Keep in mind that the Python agent doesn't capture telemetry for Azure Functions without our integration. We recommend installing the [Azure Functions monitoring integration](/docs/infrastructure/microsoft-azure-integrations/azure-integrations-list/azure-functions-monitoring-integration/) if you'd like to collect data about Azure Functions.
+
+## Install the Python agent onto Azure App Service or Container Apps[#install-python]
+
+To add the New Relic Python agent to your Azure application, add `newrelic` to your `requirements.txt` file.
+
+Here's an example with a sample app:
+
+```python
+# app.py
+# Run with command "flask run"
+
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route('/')
+def hello_world():
+    return "Hello, fellow humans!"
+
+if __name__ == '__main__':
+    app.run()
+```
+
+On the command line:
+
+```console
+foo@bar:~$ cat requirements.txt 
+newrelic
+flask
+```
+
+If you're using a Dockerfile, add `newrelic-admin run-program` as an entrypoint to your startup script.
+
+```lang-dockerfile
+COPY requirements.txt .
+COPY app.py .
+
+RUN pip install -r requirements.txt
+
+EXPOSE 5000
+
+ENTRYPOINT ["newrelic-admin", "run-program"]
+CMD ["flask", "run"]
+```
+
+Otherwise, initialize the agent by adding these lines to your app:
+
+```python
+# app.py
+# Run with command "flask run"
+
+import newrelic.agent   # ADDED
```

Comment:
Is initializing the same as adding the Python agent?

Reply:
Sort of--the way the agent works is that you install the agent on the host (in this case, Azure), but you have to do something to let the app know that it needs to run the code with New Relic.  This is done either with the New Relic commands in front of the startup script or the initialization function.

TL;DR: I wanted to distinguish between installing New Relic and actually flagging New Relic to run on the app

Reply:
Ah ok, and you install New Relic Python agent to the Azure app by running a bash command, "adding" New Relic to the flask app